ql/src/java/org/apache/hadoop/hive/ql/parse/QBParseInfo.java

```diff
@@ -211,8 +211,13 @@ public boolean isDestToOpTypeInsertOverwrite(String clause) {
   /**
    * See also {@link #getInsertOverwriteTables()}
    */
-  public boolean isInsertIntoTable(String dbName, String table) {
-    String fullName = dbName + "." + table;
+  public boolean isInsertIntoTable(String dbName, String table, String branchName) {
+    String fullName;
+    if (branchName != null) {
```

Comment:
could we have some helper for that to avoid duplication in multiple places:

Reply:
Done. It looks much clean now. :)

Reply:
````
String  fullName = dbName + "." + table;
if (branchName != null) {
    fullName += "." + branchName;
}
````

Reply:
Done. Thanks.